### PR TITLE
gnumeric 1.10.17

### PR DESCRIPTION
**Pull request by ghost**
I have alot of gnumeric files.

Did not want to update libgsf: Note about last version of python module.

Therefore goffice had to stay where it's at, thus gnumeric 1.10.17

This build GTK only --without-perl.

Tested samples and 119 other files.

Very, Very impressed with MATE desktop opening all at once.

*************** RESULTS ***************
1033	WARNING Invalid attribute VAlign expected integer received GNM_VALIGN_BOTTOM
1004	WARNING Invalid attribute HAlign expected integer received GNM_HALIGN_GENERAL
50	WARNING Invalid attribute value expected integer received GNM_PRINT_ACTIVE_SHEET
44	WARNING Invalid attribute VAlign expected integer received GNM_VALIGN_TOP
41	WARNING Invalid attribute HAlign expected integer received GNM_HALIGN_LEFT
23	WARNING Invalid attribute HAlign expected integer received GNM_HALIGN_CENTER
6	WARNING Invalid attribute HAlign expected integer received GNM_HALIGN_CENTER_ACROSS_SELECTION
5	WARNING Invalid attribute HAlign expected integer received GNM_HALIGN_RIGHT
2	WARNING Invalid attribute VAlign expected integer received GNM_VALIGN_CENTER
2	WARNING Invalid attribute value expected integer received GNM_PRINT_ALL_SHEETS
1	WARNING Invalid attribute cols expected integer received 0.000000
1	WARNING Invalid attribute Style expected integer received GNM_VALIDATION_STYLE_NONE
1	WARNING Invalid attribute Type expected integer received GNM_VALIDATION_TYPE_IN_LIST
	
2213	Totals from 119 files downloaded off the net. (Batched one at a time and Opened all at once in Caja)
	
	
4	GLib-CRITICAL Source ID XXX was not found when attempting to remove it
	Note: this appears to be unfixed GTK bug.

### CODE REVIEW

**Comment by pyhalov** [comment]
Can you investigate what depends on libgsf python modules? I'd like to avoid shipping gnumeric dated back to 2011...

**Comment by pyhalov** [comment]
I  that see gnome-commander and goffice depend on libgsf. But it seems they don't include python code. Perhaps, just bump libgsf and recompile dependencies if necessary?

**Comment by ghost** [comment]
Will try to update all these to latest.

On Wed, Aug 23, 2017 at 2:01 PM, Alexander Pyhalov <notifications@github.com

**Comment by ghost** [comment]
All at latest. Most all errors went away. No tests done on libgsf. Made them work on goffice. Gnumeric tests would not work right. Needs more time to understand. Tested both 32 and 64 bit gnumeric on same 150 plus sheets. Had one core using python console plugin in 64 bit gnumeric the first time but I confused it. Not really a Python or C programmer. One patch in goffice, works for me. Github and command line git new to me. Sorry if I missed something. Tired and don't feel good.

File: components/goffice/Makefile [hunk 1]
```diff
@@ -10,20 +10,21 @@
```
Comment (pyhalov) [inline components/goffice/Makefile:10]: Please, move it to appropriate category (library) http://wiki.openindiana.org/oi/Categories+layout

File: components/goffice/Makefile [hunk 2]
```diff
@@ -10,20 +10,21 @@
...
+# Copyright 2017 Jason Martin
 #
 
 include ../../make-rules/shared-macros.mk
 
 COMPONENT_NAME=		goffice
-COMPONENT_VERSION=	0.8.17
-COMPONENT_REVISION=	1
+COMPONENT_VERSION=	0.10.35
+COMPONENT_REVISION=	0
```
Comment (pyhalov) [inline components/goffice/Makefile]: COMPONENT_REVISION = 0 should be omitted (from all components)

File: components/goffice/patches/go-math.patch [hunk 3]
```diff
@@ -0,0 +1,15 @@
+--- goffice-0.10.35/goffice/math/go-math.c	Wed Apr 15 18:43:21 2015
++++ goffice-0.10.35/goffice/math/go-math.c.new	Thu Aug 24 09:35:18 2017
+@@ -313,7 +313,11 @@
+ {
+ 	/* What a circus!  */
+ #ifdef HAVE_FINITE
+-	return finite (x);
```
Comment (pyhalov) [inline components/goffice/patches/go-math.patch]: Why can't we use finite()? Perhaps, just rearrage #ifdefs to make it less confusing

File: components/library/libgsf/Makefile [hunk 4]
```diff
@@ -11,19 +11,20 @@
...
+# Copyright 2017 Jason Martin
 #
 
 include ../../../make-rules/shared-macros.mk
 
 COMPONENT_NAME= libgsf
-# This is the last version which provides python gsfgnomemodule
-COMPONENT_VERSION= 1.14.21
-COMPONENT_REVISION= 2
+# 1.14.21 was the last version which provides python gsfgnomemodule
```
Comment (pyhalov) [inline components/library/libgsf/Makefile]: remove comment

**Comment by ghost** [comment]
Adjusted. Could test libgsf but 2 tests for Valgrind still failed. Seven passed. Found better solution in a dicussion for blender on Solaris.

**Comment by ghost** [comment]
Should have said, found better patch for go-math.h in goffice.
Gnumeric tests are at 46 pass, 47 skip and 44 fail but there is a problem in current test scripts where it does not use zcat on hipster correctly (23 fails). Some are problems where a capital Z gets tacked onto the end of filenames and the last are valgrind tests failing like in libgsf.

**Comment by ghost** [comment]
Here is zcat problem in gnumeric tests and zcat is also where my capital Z comes from in other tests
GnumericTest::test_importer(GnumericTest.pm:400):
400:        my $htxt = `zcat -f '$tmp' | $normalize_gnumeric | sha1sum`;
  DB<1> s
Invalid Option
Usage: zcat [file ...]

**Comment by pyhalov** [comment]
Patch test suite

File: components/desktop/gnumeric/Makefile [hunk 5]
```diff
@@ -44,17 +44,17 @@ CONFIGURE_ENV+= PYTHON=$(PYTHON)
 COMPONENT_INSTALL_ENV+= GCONF_DISABLE_MAKEFILE_SCHEMA_INSTALL=1
 
 # common targets
-build:		$(BUILD_32)
+build:		$(BUILD_32_and_64)
```
Comment (pyhalov) [inline components/desktop/gnumeric/Makefile]: Why do we need both versions? If 64-bit version is sane, deliver just it.

**Comment by ghost** [comment]
Ready.

File: components/desktop/gnumeric/gnumeric.p5m [hunk 6]
```diff
@@ -0,0 +1,3787 @@
...
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright 2017 <contributor>
```
Comment (pyhalov) [inline components/desktop/gnumeric/gnumeric.p5m]: Add your copyright here

**Comment by pyhalov** [comment]
please, split this PR in several: 1) libgsf/goffice update (do they depend on each other? If they do, they should be updated in different PRs), 2) bump dependent component's COMPONENT_REVISION, so that they would be rebuilt, 3) Add gnumeric.

**Comment by ghost** [comment]
I fixed copyright and added COMPONENT_REVISION to goffice.
Is there a document that explains how things get rebuild using COMPONENT_REVISION?
When I do new PR's, do I need to create new branchs per PR?
Do you close this one?

**Comment by pyhalov** [comment]
Yes, you'll likely want to have a branch per PR. The components will be rebuilt when Makefile is changed. The 'intelligent' way to trigger rebuild is to bump COMPONENT_REVSION of dependent components.

**Comment by pyhalov** [comment]
Superseded by https://github.com/OpenIndiana/oi-userland/pull/3454

**Pull request closed without merging**